lib/logstash/inputs/twitter.rb

```diff
@@ -101,10 +104,14 @@ class LogStash::Inputs::Twitter < LogStash::Inputs::Base
   def register
     require "twitter"
 
-    if !@use_samples && ( @keywords.nil? && @follows.nil? && @locations.nil? )
+    if !@use_samples && !@use_firehose && ( @keywords.nil? && @follows.nil? && @locations.nil? )
       raise LogStash::ConfigurationError.new("At least one parameter (follows, locations or keywords) must be specified.")
     end
 
+    if @use_samples && @use_firehose
+      raise LogStash::ConfigurationError.new("Only use_firehose or use_samples can be choose")
```

Comment:
Typo on `choose` - maybe rephrase "use_firehose and use_samples cannot both be true. Choose one or the other." ?